### ISSUE TITLE
no-empty-pattern should allow on rest

### ISSUE THREAD

**Issue by YairRand** [eslint/eslint#11309]
**What rule do you want to change?**
`no-empty-pattern`

**Does this change cause the rule to produce more or fewer warnings?**
Fewer.

**How will the change be implemented? (New option, new default behavior, etc.)?**
New default behavior.

**Please provide some example code that this change will affect:**

```js
var object = { a: 1, b: 2, c: 3 },
  { c: {}, ...allPropertiesExceptC } = object;
```
The above pattern is useful for preventing certain properties from being copied to the object, without creating new variables unnecessarily.

**What does the rule currently do for this code?**
Even when the destructuring assignment contains a rest operator (`...`), ESLint shows an error because of the `c: {}`. `no-empty-pattern` makes sense in normal situations like `var {a: {}} = foo;` where there's no reason for the assignment to have an empty pattern that sets nothing, but when the rest operator is present, this behavior does not make sense.

**What will the rule do after it's changed?**
`no-empty-pattern` should only flag situations where an empty pattern is used in a destructuring that does not include a rest operator.

**Are you willing to submit a pull request to implement this change?**
No, sorry.

**Comment by nzakas** [comment]
Thanks for the report. The purpose of the rule is to flag for any empty pattern regardless of what else occurs in the pattern, so I believe this is the correct default behavior.

I could see potentially adding an option to opt-in to the behavior you describe, though I'd like to see if there's support on the team for doing that.

**Comment by ljharb** [comment]
Given there’s an ignore destructuring option on no-unused-vars, this does seem like an interesting option.

Note that in practice, this will throw when the property is null or undefined.
